src/vulkan_pnext_chain.cpp

```diff
@@ -0,0 +1,188 @@
+	debugLabel.pLabelName = "vr-marker,frame_end,type,application";
+	debugLabel.color[0] = 1.f;
+	debugLabel.color[1] = 0.f;
+	debugLabel.color[2] = 0.f;
+	debugLabel.color[3] = 1.f;
+
+	PFN_vkCmdInsertDebugUtilsLabelEXT pfnVkCmdInsertDebugUtilsLabelEXT = (PFN_vkCmdInsertDebugUtilsLabelEXT) vkGetInstanceProcAddr(vulkan.instance, "vkCmdInsertDebugUtilsLabelEXT");
+	assert(pfnVkCmdInsertDebugUtilsLabelEXT != nullptr);
+	(*pfnVkCmdInsertDebugUtilsLabelEXT)(commandBuffer, &debugLabel);
+
+	result = vkEndCommandBuffer(commandBuffer);
+	assert(result == VK_SUCCESS);
+
+	VkFenceCreateInfo fenceCreateInfo;
+	fenceCreateInfo.sType = VK_STRUCTURE_TYPE_FENCE_CREATE_INFO;
+	fenceCreateInfo.pNext = nullptr;
+	fenceCreateInfo.flags = 0;
+
+	VkFence fence;
+	result = vkCreateFence(vulkan.device, &fenceCreateInfo, nullptr, &fence);
+	assert(result == VK_SUCCESS);
+
+	VkSubmitInfo submitInfo;
+	submitInfo.sType = VK_STRUCTURE_TYPE_SUBMIT_INFO;
+	submitInfo.pNext = nullptr;
+	submitInfo.waitSemaphoreCount = 0;
+	submitInfo.pWaitSemaphores = nullptr;
+	submitInfo.pWaitDstStageMask = nullptr;
+	submitInfo.commandBufferCount = 1;
+	submitInfo.pCommandBuffers = &commandBuffer;
+	submitInfo.signalSemaphoreCount = 0;
+	submitInfo.pSignalSemaphores = nullptr;
+
+	// Create image and associated memory
+
+	constexpr uint32_t queueFamilyIndex = 0;
+	VkImageCreateInfo imageCreateInfo;
+	imageCreateInfo.sType = VK_STRUCTURE_TYPE_IMAGE_CREATE_INFO;
+	imageCreateInfo.pNext = nullptr;
+	imageCreateInfo.flags = 0;
+	imageCreateInfo.imageType = VK_IMAGE_TYPE_2D;
+	imageCreateInfo.format = VK_FORMAT_R8G8B8A8_UNORM;
+	imageCreateInfo.extent.width = 192;
+	imageCreateInfo.extent.height = 108;
+	imageCreateInfo.extent.depth = 1;
+	imageCreateInfo.mipLevels = 1;
+	imageCreateInfo.arrayLayers = 1;
+	imageCreateInfo.samples = VK_SAMPLE_COUNT_1_BIT;
+	imageCreateInfo.tiling = VK_IMAGE_TILING_OPTIMAL;
+	imageCreateInfo.usage = VK_IMAGE_USAGE_TRANSFER_DST_BIT | VK_IMAGE_USAGE_TRANSFER_SRC_BIT;
+	imageCreateInfo.sharingMode = VK_SHARING_MODE_EXCLUSIVE;
+	imageCreateInfo.queueFamilyIndexCount = 1;
+	imageCreateInfo.pQueueFamilyIndices = &queueFamilyIndex;
+	imageCreateInfo.initialLayout = VK_IMAGE_LAYOUT_TRANSFER_DST_OPTIMAL;
+
+	VkImage image;
+	result = vkCreateImage(vulkan.device, &imageCreateInfo, nullptr, &image);
+	assert(result == VK_SUCCESS);
+
+	VkMemoryRequirements memoryRequirements;
+	vkGetImageMemoryRequirements(vulkan.device, image, &memoryRequirements);
+
+	VkMemoryAllocateInfo memoryAllocateInfo;
+	memoryAllocateInfo.sType = VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO;
+	memoryAllocateInfo.pNext = nullptr;
+	memoryAllocateInfo.allocationSize = memoryRequirements.size;
+	memoryAllocateInfo.memoryTypeIndex = 0;
+
+	VkDeviceMemory deviceMemory;
+	result = vkAllocateMemory(vulkan.device, &memoryAllocateInfo, nullptr, &deviceMemory);
+	assert(result == VK_SUCCESS);
+
+	/* Where the magic happens...
+
+	Those lines check that the trace tool does not save and reuse application owned memory
```

Comment:
I would remove the second sentence since it would hopefully become false soon.

Reply:
Removed, and changed a bit the comment to make it more explicit.